OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -88,19 +103,29 @@ void ITick.Tick(Actor self)
 			if (info.Delay > 0 && --tick == 0)
 				PlayAttackAnimation(self);
 
-			if (string.IsNullOrEmpty(info.AimSequence) && string.IsNullOrEmpty(info.ReloadPrefix))
+			if (onlyAttackAnim)
 				return;
 
-			var sequence = wsb.Info.Sequence;
-			if (!string.IsNullOrEmpty(info.AimSequence) && attack.IsAttacking)
-				sequence = info.AimSequence;
+			if (attackAnimPlaying)
+				return;
 
-			var prefix = (armament.IsReloading && !string.IsNullOrEmpty(info.ReloadPrefix)) ? info.ReloadPrefix : "";
+			var enabledBodies = wsbs.Where(Exts.IsTraitEnabled);
+			if (!enabledBodies.Any())
```

Comment:
This early abort probably does not even abort early since the `foreach` loop would find nothing and end without doing anything anyways.